Sources/MapboxMaps/Location/Pucks/Puck2D.swift

```diff
@@ -184,11 +189,15 @@ internal extension Puck2D {
         layer.topImageSize = configuration.resolvedScale
         layer.bearingImageSize = configuration.resolvedScale
         layer.shadowImageSize = configuration.resolvedScale
-        layer.accuracyRadius = .constant(location.horizontalAccuracy)
         layer.emphasisCircleRadiusTransition = StyleTransition(duration: 0, delay: 0)
         layer.bearingTransition = StyleTransition(duration: 0, delay: 0)
-        layer.accuracyRadiusColor = .constant(ColorRepresentable(color: UIColor(red: 0.537, green: 0.812, blue: 0.941, alpha: 0.3)))
-        layer.accuracyRadiusBorderColor = .constant(ColorRepresentable(color: .lightGray))
+
+        // Horizontal accuracy ring is an optional visual for the 2D Puck
+        if configuration.showAccuracyRing {
+            layer.accuracyRadius = .constant(location.horizontalAccuracy)
+            layer.accuracyRadiusColor = .constant(ColorRepresentable(color: UIColor(red: 0.537, green: 0.812, blue: 0.941, alpha: 0.3)))
+            layer.accuracyRadiusBorderColor = .constant(ColorRepresentable(color: .lightGray))
+        }
```

Comment:
Do we need to set different values in the else condition?

Reply:
no, if we don't want to show the accuracy ring, we don't want to set any values at all